respy/exogenous_processes.py

```diff
@@ -0,0 +1,102 @@
+"""This module contains everything related to exogenous processes.
+
+In models without exogenous processes the law of motion, the transition of states over
+periods, is deterministic. Given a state and a choice the state in the next period is
+certain.
+
+Exogenous processes introduce a stochastic element to the law of motion. Given a state
+and a choice there exist multiple states in the next period which are valid successors.
+The transition to one of the successors is determined by a probability which depends on
+the characteristics of the state.
+
+See Also
+--------
+respy.pre_processing.model_processing._parse_exogenous_processes
+respy.pre_processing.model_processing._parse_observables
+respy.state_space._MultiDimStateSpace._weight_continuation_values
+respy.simulate._apply_law_of_motion
+
+"""
+import itertools
+
+import numpy as np
+from scipy import special
+
+from respy.config import COVARIATES_DOT_PRODUCT_DTYPE
+from respy.parallelization import parallelize_across_dense_dimensions
+from respy.shared import create_dense_state_space_columns
+
+
+@parallelize_across_dense_dimensions
+def compute_transition_probabilities(states, optim_paras):
```

Comment:
In general this is a nice and easy way to allow for great flexibility in defining exogenous processes. I really like the implementation!